main.nf

```diff
@@ -1,77 +1,71 @@
 // Ensure DSL2
 nextflow.enable.dsl = 2
 
-process SYNAPSE_GET {
+//path to exmaple csv
+params.csv_path = "data/input.csv"
 
-    container "sagebionetworks/synapsepythonclient:v2.7.0"
+//checks metadata, and passes relavent fields along through .json
+process SYNAPSE_CHECK {
+        
+    container "python:3.10.4"
     
     secret 'SYNAPSE_AUTH_TOKEN'
 
     input:
-    val syn_id
+    tuple val(syn_id), val(md5_checksum)
 
     output:
-    tuple val(syn_id), path('*')
+    tuple val(syn_id), val(md5_checksum), path('*file_entity_test.json')
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** Since you'll only ever have one JSON file, you can just glob `*.json`.